transport/handler.go

```diff
@@ -22,7 +22,38 @@ package transport
 
 import "golang.org/x/net/context"
 
-// Handler handles a single transport-level request.
+// Mode is an enum of RPC types
+type Mode int
+
+const (
+	// Unknown RPC type
+	Unknown = iota
+	// Unary RPC type
+	Unary = iota
+	// Oneway RPC type
+	Oneway
+)
+
+func (m Mode) String() string {
+	switch m {
+	case Unary:
+		return "Unary"
+	case Oneway:
+		return "Oneway"
+	default:
+		return "Unknown"
+	}
+}
+
+// HandlerInfo holds a handler and its mode
+type HandlerInfo struct {
```

Comment:
Let's call this `HandlerSpec` since it contains a spec that the transport must inspect and call. The variable at rest should be called `spec` like so:

```
spec := h.Registry.GetHandlerSpec(...)
```